axiom/tests/test_log_event.c

```diff
@@ -0,0 +1,359 @@
+
+  nr_log_event_destroy(&event);
+}
+
+static void test_log_event_log_level(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  /*
+   * Test : Bad parameters.
+   */
+  nr_log_event_set_log_level(event, NULL);
+  tlib_pass_if_null("NULL log_level", event->log_level);
+  nr_log_event_set_log_level(NULL, "test log_level");
+  tlib_pass_if_null("NULL event", event->log_level);
+
+  /*
+   * Test : Valid log_level.
+   */
+  nr_log_event_set_log_level(event, "test log_level");
+  tlib_pass_if_str_equal("Valid log_level set", "test log_level",
+                         event->log_level);
+  nr_log_event_set_log_level(event, "another test log_level");
+  tlib_pass_if_str_equal("Another valid log_level set",
+                         "another test log_level",
+                         event->log_level);
+
+  nr_log_event_destroy(&event);
+}
+
+static void test_log_event_hostname(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  /*
+   * Test : Bad parameters.
+   */
+  nr_log_event_set_hostname(event, NULL);
+  tlib_pass_if_null("NULL hostname", event->hostname);
+  nr_log_event_set_hostname(NULL, "test hostname");
+  tlib_pass_if_null("NULL event", event->hostname);
+
+  /*
+   * Test : Valid hostname.
+   */
+  nr_log_event_set_hostname(event, "test hostname");
+  tlib_pass_if_str_equal("Valid hostname set", "test hostname",
+                         event->hostname);
+  nr_log_event_set_hostname(event, "another test hostname");
+  tlib_pass_if_str_equal("Another valid hostname set",
+                         "another test hostname",
+                         event->hostname);
+
+  nr_log_event_destroy(&event);
+}
+
+static void test_log_event_timestamp(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  // Test : Get timestamp with a NULL event
+  tlib_pass_if_time_equal("NULL event should give zero", 0,
+                          event->timestamp);
+
+  // Test : Set the timestamp a couple times
+  nr_log_event_set_timestamp(event, 553483260);
+  tlib_pass_if_time_equal("Get timestamp should equal 553483260",
+                          553483260 / NR_TIME_DIVISOR_MS,
+                          event->timestamp);
+  nr_log_event_set_timestamp(event, 853483260);
+  tlib_pass_if_time_equal("Get timestamp should equal 853483260",
+                          853483260 / NR_TIME_DIVISOR_MS,
+                          event->timestamp);
+
+  nr_log_event_destroy(&event);
+}
+
+static void test_log_event_span_id(void) {
+  nr_log_event_t* event = nr_log_event_create();
+
+  // Test : that is does not blow up when we give the setter a NULL pointer
+  nr_log_event_set_span_id(event, NULL);
+  nr_log_event_set_span_id(NULL, "wallaby");
+  tlib_pass_if_null("the span should still be NULL",
+                    event->span_id);
+
+  // Test : the getter should not blow up when we send it an event with a NULL
+  // spanID
+  tlib_pass_if_null("NULL event -> NULL span ID",
+                    event->span_id);
```

Comment:
redundant

Reply:
Thanks, addressed in https://github.com/newrelic/newrelic-php-agent/pull/490/commits/ea9a50a125729a1422335e5b182d676e19a953c5